CONTRIBUTING.md

```diff
@@ -169,41 +169,68 @@ npm install abbrev -w ripple-keypairs
-11. Run `npx lerna publish from-package --yes` - This will actually publish the packages.
-  * NOTE: if you're releasing a beta, run `npx lerna publish from-package --dist-tag beta --yes` instead.
-  * If it asks for it, enter your [npmjs.com](https://npmjs.com) OTP (one-time password) to complete publication.
-12. Create a new branch (`git checkout -b <BRANCH_NAME>`)to capture the updated packages from the release. Merge those changes into `main`. (You can skip this step on a beta release).
+## Release process + checklist
+
+## Pre-release
+- [ ] Your changes should be on a branch.
+- [ ] Your changes should have unit tests.
+- [ ] Lint the code with `npm lint`
+- [ ] Build your code with `npm build`
+- [ ] Run the unit tests with `npm test`
+- [ ] Get a full code review.
+- [ ] Merge your branch into `main` and push to github.
+- [ ] Ensure that all tests passed on the last CI that ran on `main`.
+
+## Release
+
+- [ ] Checkout `main` (or your beta branch) and `git pull`.
+- [ ] Create a new branch (`git checkout -b <BRANCH_NAME>`) to capture updates that take place during this process.
+- [ ] Update `HISTORY.md` to reflect release changes.
+  - [ ] Update the version number and release date, and ensure it lists the changes since the previous release.
+- [ ] Run `npm run docgen` if the docs were modified in this release to update them (skip this step for a beta).
+- [ ] Run `npm run build` to triple check the build still works
+- [ ] Run `npx lerna version --no-git-tag-version` - This creates a draft PR and bumps the versions of the packages.
+  - [ ] For each changed package, pick what the new version should be. Lerna will bump the versions, commit version bumps to `main`, and create a new git tag for each published package.
+  - [ ] If publishing a beta, make sure that the versions are all of the form `a.b.c-beta.d`, where `a`, `b`, and `c` are identical to the last normal release except for one, which has been incremented by 1.
+- [ ] Run `npm i` to update the package-lock with the updated versions.
+- [ ] Create a new PR from this branch into `main` and merge it (you can directly merge into the beta branch for a beta).
+- [ ] Checkout `main` and `git pull` (you can skip this step for a beta since you already have the latest version of the beta branch).
+- [ ] Actually publish the packages with one of the following:
+  - [ ] Stable release: Run `npx lerna publish from-package --yes`
+  - [ ] Beta release: Run `npx lerna publish from-package --dist-tag beta --yes`
+        Notice this allows developers to install the package with `npm add xrpl@beta`
+  - [ ] If requested, enter your [npmjs.com](https://npmjs.com) OTP (one-time password) to complete publication.
+- [ ] If not a beta release: Create a new branch (`git checkout -b <BRANCH_NAME>`) to capture the updated packages from the release. Merge those changes into `main`.
```

Comment:
Re-adding numbers using auto numbering since checkboxes aren't dynamic, and with all the alt-tabbing needed it's easy to lose track of where you are unless you have the numbers. 
```suggestion
1. Checkout `main` (or your beta branch) and `git pull`.
1. Create a new branch (`git checkout -b <BRANCH_NAME>`) to capture updates that take place during this process.
1. Update `HISTORY.md` to reflect release changes.

- [ ] Update the version number and release date, and ensure it lists the changes since the previous release.

1. Run `npm run docgen` if the docs were modified in this release to update them (skip this step for a beta).
1. Run `npm run build` to triple check the build still works
1. Run `npx lerna version --no-git-tag-version` - This creates a draft PR and bumps the versions of the packages.

- [ ] For each changed package, pick what the new version should be. Lerna will bump the versions, commit version bumps to `main`, and create a new git tag for each published package.
- [ ] If publishing a beta, make sure that the versions are all of the form `a.b.c-beta.d`, where `a`, `b`, and `c` are identical to the last normal release except for one, which has been incremented by 1.

1. Run `npm i` to update the package-lock with the updated versions.
1. Create a new PR from this branch into `main` and merge it (you can directly merge into the beta branch for a beta).
1. Checkout `main` and `git pull` (you can skip this step for a beta since you already have the latest version of the beta branch).
1. Actually publish the packages with one of the following:

- [ ] Stable release: Run `npx lerna publish from-package --yes`
- [ ] Beta release: Run `npx lerna publish from-package --dist-tag beta --yes`
      Notice this allows developers to install the package with `npm add xrpl@beta`
- [ ] If requested, enter your [npmjs.com](https://npmjs.com) OTP (one-time password) to complete publication.

1. If not a beta release: Create a new branch (`git checkout -b <BRANCH_NAME>`) to capture the updated packages from the release. Merge those changes into `main`.
```